app/src/main/java/at/aau/edu/wizards/ui/gameboard/recycler/GameBoardAdapter.kt

```diff
@@ -4,10 +4,14 @@ import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
 import at.aau.edu.wizards.databinding.ItemCardBinding
-import at.aau.edu.wizards.ui.gameboard.Cards
+import at.aau.edu.wizards.gameModel.GameModel
+import at.aau.edu.wizards.gameModel.GameModelCard
+import at.aau.edu.wizards.ui.gameboard.GameBoardViewModel
 
-class GameBoardAdapter : ListAdapter<Cards, GameBoardItemViewHolder>(DiffUtlCallback()) {
+class GameBoardAdapter(val viewModel: GameBoardViewModel, val model: GameModel) :
```

Comment:
usually we should keep adapter dumb, passing ViewModel is not a good approach, worst case can be memory leak too. But I don't care anymore, if it works it works - we are not building a software which is going to be used by someone.